plutus-core/plutus-core/src/PlutusCore/Builtin/KnownTypeAst.hs

```diff
@@ -271,13 +271,15 @@ instance
 
 -- Utils
 
+-- | Insert @x@ into @xs@ unless it's already there.
+type Insert :: forall a. a -> [a] -> [a]
+type family Insert x xs where
+    Insert x '[]      = '[x]
+    Insert x (x : xs) = x ': xs
+    Insert x (y : xs) = y ': Insert x xs
+
 -- | Delete all @x@s from a list.
 type family Delete x xs :: [a] where
     Delete _ '[]       = '[]
-    Delete x (x ': xs) = Delete x xs
+    Delete x (x ': xs) = xs
     Delete x (y ': xs) = y ': Delete x xs
```

Comment:
Now that we do this kind of logic in `Insert`, we don't need to do it here.

Reply:
Worth a comment? Or updating the Haddock? "Delete the first instance of `x` from a list (which is okay since we only ever put things in once"?